optaweb-vehicle-routing-frontend/src/components/SearchBox.test.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { shallow } from 'enzyme';
+import toJson from 'enzyme-to-json';
+import { OpenStreetMapProvider, SearchResult } from 'leaflet-geosearch';
+import * as React from 'react';
+import SearchBox, { IProps, IState } from './SearchBox';
+
+jest.mock('leaflet-geosearch');
+jest.useFakeTimers();
+
+beforeEach(() => {
+  (OpenStreetMapProvider as unknown as jest.MockInstance<OpenStreetMapProvider>).mockReset();
+});
+
+describe('Search box', () => {
+  it('should render text input initially', () => {
+    const props: IProps = {
+      addHandler: jest.fn(),
+      searchDelay: 1,
+    };
+    const searchBox = shallow(<SearchBox {...props} />);
+    expect(toJson(searchBox)).toMatchSnapshot();
+  });
+
+  it('should show results when query is entered', async () => {
```

Comment:
Suggestion - this test declares that "should show results when query is entered", but it focuses also on a negative scenario - empty query shows nothing. Could you consider splitting the test?

Reply:
Valid suggestion. I was afraid (lazy) of having to duplicate a lot of code in order to get the component to a state with non-empty results before issuing an empty query to be able to verify that the old results are erased. I take another look and found out it's easy to do.